src/components/context/data/DataState.ts

```diff
@@ -30,23 +30,31 @@ interface MeasurementBase {
   id: string;
   title?: string;
   instrument?: Instrument;
-  meta: Record<string, string>;
-  info: Record<string, string | number>;
+  meta: Record<string, string | number | undefined>;
+  info: Record<string, string | number | undefined>;
   data: {
     variables: Record<string, MeasurementVariable>;
   }[];
 }
 
 export type MeasurementKind = keyof Measurements;
 
-export const kindsLabel: Record<MeasurementKind, string> = {
+export const kindLabelMap: Record<MeasurementKind, string> = {
   ir: 'IR',
   raman: 'Raman',
   uv: 'UV',
   mass: 'Mass',
   nmr1h: 'NMR 1H',
   other: 'Other',
 };
+export const kindsLabel: MeasurementKind[] = [
```

Comment:
Please don't create this new variable and when you need the properties just use `Object.keys(kindLabel)`. It should not be a limiting speed step. And even we we wanted to create this second array we should not write all the values but also use `Object.keys`.

Reply:
fixed ,
thank you